riff-raff/app/notification/FailureNotificationContents.scala

```diff
@@ -0,0 +1,61 @@
+package notification
+
+import com.gu.anghammarad.models.{Action, Target}
+import controllers.routes
+import deployment.{NoDeploysFoundForStage, ScheduledDeployError, SkippedDueToPreviousFailure, SkippedDueToPreviousWaitingDeploy}
+import magenta.DeployParameters
+
+import java.util.UUID
+
+object FailureNotificationContents {
+
+  case class NotificationContents(subject: String, message: String, actions: List[Action])
+  case class NotificationContentsWithTargets(notificationContents: NotificationContents, targets: List[Target])
+
+  def problematicDeployUrl(prefix: String, uuid: UUID): String = {
+    val path = routes.DeployController.viewUUID(uuid.toString, verbose = true)
+    prefix + path.url
+  }
+
+  def deployAgainUrl(prefix: String, uuid: UUID): String = {
+    val path = routes.DeployController.deployAgainUuid(uuid.toString)
+    prefix + path.url
+  }
+
+  def scheduledDeployConfigUrl(prefix: String): String = {
+    val path = routes.ScheduleController.list()
+    prefix + path.url
+  }
+
+  def viewProblematicDeploy(uuid: UUID, status: String, urlPrefix: String) = Action(s"View $status deploy", problematicDeployUrl(urlPrefix, uuid))
+
+  def deployFailedNotificationContents(uuid: UUID, parameters: DeployParameters, urlPrefix: String): NotificationContents = {
+    NotificationContents(
+      subject = s"${parameters.deployer.name} failed",
+      message = s"${parameters.deployer.name} for ${parameters.build.projectName} (build ${parameters.build.id}) to stage ${parameters.stage.name} failed.",
+      actions = List(viewProblematicDeploy(uuid, "failed", urlPrefix))
+    )
+  }
+
+  def deployUnstartedNotificationContents(scheduledDeployError: ScheduledDeployError, urlPrefix: String, teamTargetsSearch: (Option[UUID], Option[DeployParameters]) => List[Target], riffRaffTargets: List[Target]): NotificationContentsWithTargets = {
```

Comment:
We pass in a function for `teamTargetsSearch`, which makes it easy to use a mock function in unit tests:

`def fakeGetTargets(uuid: Option[UUID], maybeParameters: Option[DeployParameters]): List[Target] = List(Stack("another-team-stack"))`

This is necessary as the real `getTargets` function talks to s3 etc.